include/vcpkg/base/message-data.inc.h

```diff
@@ -167,6 +167,13 @@ DECLARE_MESSAGE(ARegistryPathMustStartWithDollar,
                 "",
                 "A registry path must start with `$` to mean the registry root; for example, `$/foo/bar`.")
 DECLARE_MESSAGE(ARelaxedVersionString, (), "", "a relaxed version string")
+DECLARE_MESSAGE(ArtifactsBootstrapFailed, (), "", "vcpkg-artifacts is not installed and could not be bootstrapped.")
+DECLARE_MESSAGE(ArtifactsNotInstalledReadonlyRoot,
+                (),
+                "",
+                "vcpkg-artifacts is not installed, and it can't be installed because VCPKG_ROOT is assumed to be "
+                "readonly. Reinstalling vcpkg using the 'one liner' may fix this problem.")
```

Comment:
1. Maybe, but if the user ever sees this something about the world is broken. Because the only 'readonly' case right now is VS and artifacts should be deployed by the VS installer there. If anything it might make sense to go the other way and replace this with a generic 'internal error' message.
2. Such as.... ?